### ISSUE TITLE
Function to alter the volume of a note associated with a specific voice

### ISSUE THREAD

**Issue by habuenav** [shorepine/amy#253]
Hi team, I don't know if this function already exists, if it does I don't know how to do it so I would appreciate if you could help me with that, otherwise my proposal is the following.

I would like to propose the addition of a function that allows to alter the volume of a note associated with a specific voice without having to create a struct event e = amy.default_event(); and then having to do an amy.add_event(e);.
To change the volume of a note, I currently access the variable amy_global.volume, which performs a smooth volume transition, but affects the overall volume. I would like the volume to be able to be managed individually for each voice.

Benefits:
Improved sound quality by avoiding sudden jumps in the volume of the notes.
Greater control over the volume of each voice independently.
Simplifying the volume adjustment process for developers.

Additional comments:
I appreciate your attention and hope you will consider this proposal to improve the functionality of the project.
Thank you for your time and effort.

**Comment by dpwe** [comment]
I'm sorry this appears to not have been addressed. 

The answer *should* be that you can adjust the `amp` ControlCoefficients for a sounding note to trim its amplitude:

```
amy.reset()
# Set up a simple sine voice with an initial pluck attack and a slow release.
amy.send(osc=0, wave=amy.SINE, bp0='0,1,500,0.2,1000,0')
# Start a note
amy.send(osc=0, note=60, vel=1)
# You can adjust its amp during sustain without retriggering
amy.send(osc=0, amp=1.5)
```

This works for single oscs.  It also works for `wave=amy.ALGO` (i.e. FM/DX7) voices.  But it doesn't work for the analog synth (Juno) patches nor the additive synth (Piano, `load_patch=256`) patch.  

The reason is that these patches are constructed as multiple independent oscillators, each with their own envelopes, that are then summed.  For Juno patches, osc 0 is in fact the square wave/PWM osc, so if you change the amplitude of osc 0 you'll only affect that part of the voice.  You can trim the overall volume of the note, but only by changing all 4 of the sounding oscs (PWM, sawtooth, suboctave, and noise):

```
amy.reset()
# Set up voice 0 as a juno patch 1
amy.send(voices='0', load_patch=1)
# Start a note
amy.send(voices='0', note=60, vel=1)
# Scale the amp of all 4 sounding oscs
for osc in range(4):
    amy.send(voices='0', osc=osc, amp=1.5)
```
A similar approach would work for the Piano voice, but there you'd have to scale 21 oscs.  

This isn't a good solution; in general, there's not even a way to find out the number of oscs in a voice.  It would be better for all patches to have osc 0 as a control osc with effects that apply to the entire note.  We would define a new `wave=amy.CONTROL_OSC` or similar that simply propagates note-on/off commands, then applies envelope (and filter) to the summed result.

One problem is what happens on release: If osc 0's envelope is applied to the sum of all the component oscs, then the default osc envelope (which has a release time of zero) would cut off any nonzero release in the remaining oscs.  Instead, we would want a default control osc envelope with infinite release time, so the actual releases are determined by the individual oscs.  This doesn't seem too difficult.

**Comment by dpwe** [comment]
Hmm, it looks like I meant to fix this for additive-synthesis (`wave=amy.PARTIALS`) to address #215.  I see the dynamic amplitude of the control osc being [passed to the individual oscillators via their `velocity` field](https://github.com/shorepine/amy/blob/81c59c746627bd9687b3891ecb6b20a6bed8a519/src/partials.c#L212) -- a bit of an abuse IMO, but it would work, *except* we [turn off velocity-sensitivity](https://github.com/shorepine/amy/blob/81c59c746627bd9687b3891ecb6b20a6bed8a519/src/interp_partials.c#L77) for the interpolated partials voices (i.e., the `load_patch=256` piano).  

It would be easy to turn it on again, but I don't see any infinite-release default envelope for the parent osc, so if implemented alone, I think it would have the chopped-off release problem.  This is probably why velocity coupling is turned off for the `interp_partials` voices.

Bottom line: Fiddling with the `amp` of the control osc may work for `PARTIALS` and `BYO_PARTIALS`, but it doesn't work for `INTERP_PARTIALS` which was the most relevant case here.
